package/src/components/MessageOverlay/OverlayReactions.tsx

```diff
@@ -109,58 +84,55 @@ export type Reaction = {
 export type OverlayReactionsProps<
   StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
 > = Pick<MessageOverlayContextValue<StreamChatGenerics>, 'OverlayReactionsAvatar'> & {
-  reactions: Reaction[];
   showScreen: Animated.SharedValue<number>;
   title: string;
   alignment?: Alignment;
+  messageId?: string;
+  reactions?: Reaction[];
   supportedReactions?: ReactionData[];
 };
 
-type ReactionIconProps = Pick<Reaction, 'type'> & {
-  pathFill: string;
-  size: number;
-  supportedReactions: ReactionData[];
-};
-
-const ReactionIcon = ({ pathFill, size, supportedReactions, type }: ReactionIconProps) => {
-  const Icon = supportedReactions.find((reaction) => reaction.type === type)?.Icon || Unknown;
-  return <Icon height={size} pathFill={pathFill} width={size} />;
-};
-
 /**
  * OverlayReactions - A high level component which implements all the logic required for message overlay reactions
  */
 export const OverlayReactions = (props: OverlayReactionsProps) => {
+  const [itemHeight, setItemHeight] = React.useState(0);
   const {
     alignment: overlayAlignment,
+    messageId,
     OverlayReactionsAvatar,
-    reactions,
+    reactions: propReactions,
     showScreen,
     supportedReactions = reactionData,
     title,
   } = props;
   const layoutHeight = useSharedValue(0);
   const layoutWidth = useSharedValue(0);
-
-  const [itemHeight, setItemHeight] = React.useState(0);
+  const {
+    loading,
+    loadNextPage,
+    reactions: fetchedReactions,
+  } = useFetchReactions({
+    messageId,
+    sort: { created_at: -1 },
```

Comment:
```suggestion
    sort: { created_at: -1 },
```

I think we should move this to a const outside, so that inside the hook useMEmo stringify does not trigger again and again..

its a simple optimisation